### ISSUE TITLE
Can I use the wav2vec2 base model to train my own dataset which is of 8000 sample rate?

### ISSUE THREAD

**Issue by nome2050** [pytorch/fairseq#3236]
Can I use the wav2vec2 base model to train my own dataset which is of 8000 sample rate?

Kindly help me out?

also if yes then how much data do i need to annotate?

**Comment by alexeib** [comment]
you probably need to change the striding in feature extractor so that your representation encode approximately the same amount of audio as it would if they were 16khz. i am not sure what the best configuration is, you would have to experiment.

another option that we use more often is simply upsample your 8khz audio to 16khz (using something like sox), and then just use the official recipe for pretraining

**Comment by webbp** [comment]
It's affecting me :) I was able to [Train a wav2vec 2.0 large model](https://github.com/pytorch/fairseq/blob/main/examples/wav2vec/README.md#train-a-wav2vec-20-large-model) and [Fine-tune a pre-trained model with CTC](https://github.com/pytorch/fairseq/blob/main/examples/wav2vec/README.md#fine-tune-a-pre-trained-model-with-ctc) using fairseq-hydra-train with 8kHz wavs simply by changing all occurrences in the repo of 16000 and 16_000 to 8000 and 8_000, but

> change the striding in feature extractor

could you provide more detail about what edits to make on what files? my current guess is:
```
fairseq/models/wav2vec/wav2vec.py:56:
-default="[(512, 10, 5), (512, 8, 4), (512, 4, 2),...
+default="[(512, 10, 4), (512, 8, 3), (512, 4, 2),...
                     ^            ^
 metadata={
     "help": "convolutional feature extraction layers [(dim, kernel_size, stride), ...]"
```
and/or change dim from 512 to 256.

**Comment by egorsmkv** [comment]
The better way is just to resample audio to 16 kHz using torchaudio

**Comment by mattbonnell** [comment]
> The better way is just to resample audio to 16 kHz using torchaudio

Won't that make the forward pass much less efficient than if you were to use an 8 kHz model?
